python/GangaLSST/Lib/Im3ShapeApp/Im3ShapeRTHandler.py

```diff
@@ -0,0 +1,211 @@
+from GangaDirac.Lib.RTHandlers.DiracRTHUtils import dirac_inputdata, dirac_ouputdata, mangle_job_name, diracAPI_script_template, diracAPI_script_settings, API_nullifier, dirac_outputfile_jdl
+from GangaDirac.Lib.Files.DiracFile import DiracFile
+from Ganga.GPIDev.Lib.File.OutputFileManager import getOutputSandboxPatterns, getWNCodeForOutputPostprocessing
+from Ganga.GPIDev.Adapters.IRuntimeHandler import IRuntimeHandler
+from Ganga.GPIDev.Adapters.StandardJobConfig import StandardJobConfig
+from Ganga.Core.exceptions import ApplicationConfigurationError
+from Ganga.GPIDev.Lib.File import FileBuffer
+from Ganga.GPIDev.Lib.File.File import File
+from Ganga.Utility.Config import getConfig
+from Ganga.Utility.logging import getLogger
+from Ganga.Utility.util import unique
+from Ganga.GPIDev.Base.Proxy import isType, stripProxy
+logger = getLogger()
+
+#\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\/\#
+
+
+class Im3ShapeDiracRTHandler(IRuntimeHandler):
+
+    """The runtime handler to run the Im3ShapeApp on the Dirac backend"""
+
+    def master_prepare(self, app, appmasterconfig):
+        """
+        This function prepares the application of a master job during submit. A priori we aren't doing anything with this in Im3ShapeApp but until this is understood I'd rather not remove it
+        Args:
+            app (IApplication): This is the application given in the master job
+            appasterconfig (tuple): This is the configuration which is to prepare the app in the master job # TODO check type and this interface
+        """
+        inputsandbox, outputsandbox = master_sandbox_prepare(app, appmasterconfig)
+        return StandardJobConfig(inputbox=unique(inputsandbox),
+                outputbox=unique(outputsandbox))
+
+    def prepare(self, app, appsubconfig, appmasterconfig, jobmasterconfig):
+        """
+        This function prepares the application of the actual job being submitted, master or not
+        Args:
+            app (IApplication): This is the application actually being submitted belonging to the master or sub job being configured
+            appsubconfig (tuple): This is used to prepare the inputsandbox according to the configuration for each subjob if it varies
+            appmasterconfig (tuple): This is also used to prepare the inputsandbox but contains the config of the app for the master job
+            jobmasterconfig (StandardJobConfig): This is the configuration of the master job which may or may not be the same job as owning the app
+        """
+
+        # Construct some common objects used in job submission here
+        inputsandbox, outputsandbox = sandbox_prepare(app, appsubconfig, appmasterconfig, jobmasterconfig)
+        input_data,   parametricinput_data = dirac_inputdata(app, hasOtherInputData=True)
+
+
+        job = stripProxy(app).getJobObject()
```

Comment:
Yes but frankly this is just a nullop until it's removed... which it was in a commit which is now missing... the joys of git the history